docs/osgi.md

```diff
@@ -0,0 +1,38 @@
+# OSGI background information
+
+Some (rather incomplete) information about bundles, bundle dependencies and errors in that context.
+
+## METAINF.MF
+### Require-Bundle
+
+- specifies the explicit bundle (and optionally version) to use. If a required bundle needs to be refactored and a package moved elsewhere, then dependents will need changes to their MANIFEST.MF
+- gives you access to ALL exports of the bundle, regardless of what they are, and regardless of whether you need them. If the parts you don't need have their own dependencies you will need those to
+- bundles can be re-exported
+- although discouraged, allows the use of split packages, e.g.: a package that is spread across multiple bundles
+- can be used for non-code dependencies, e.g.: resources, Help etc.
+
+### Import-Package (preferred)
+
+- looser coupling, only the package (and optionally version) is specified and the run-time finds the required bundle
+- actual implementations can be swapped out
+- dependent packages can be moved to different bundles by the package owner
+- requires more meta data to be maintained (i.e: each package name) at lower levels of granularity
```

Comment:
This point cannot be emphasized enough. Bundles can reexport their dependencies, wheras importing packages manually you'll also need to import all transitive dependencies manually.